e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts

```diff
@@ -0,0 +1,195 @@
+        }
+
+        numRetries += 1;
+        if (numRetries > 50) {
+          fail("Chain simulator not started!");
+        }
+      }
+    } catch (e) {
+      console.error(e);
+    }
+  });
+
+  beforeEach(() => {
+    eventProcessor = new EventProcessor();
+    jest.clearAllMocks();
+  });
+
+  it('should deploy a contract and receive all the ping pong events', async () => {
+    const contractCodeRaw = fs.readFileSync('./e2e-tests/chain-simulator-e2e/contracts/ping-pong/ping-pong-egld.wasm');
+    /*
+    fn init(
+        &self,
+        ping_amount: &BigUint,
+    ) {
+     */
+    const contractArgs = [
+      '0de0b6b3a7640000', // 1 egld
+    ];
+    await fundAddress(CHAIN_SIMULATOR_URL, ALICE_ADDRESS);
+    const scAddress = await deploySc(new DeployScArgs({
+      chainSimulatorUrl: CHAIN_SIMULATOR_URL,
+      deployer: ALICE_ADDRESS,
+      contractCodeRaw: contractCodeRaw,
+      hexArguments: contractArgs,
+    }));
+    logMessage(`Deployed ping pong SC. Address: ${scAddress}`);
+
+    const numPingPongs = 20;
+
+    for (let i = 0; i < numPingPongs; i++) {
+      const pingTxHash = await sendTransaction({
+        chainSimulatorUrl: CHAIN_SIMULATOR_URL,
+        sender: ALICE_ADDRESS,
+        receiver: scAddress,
+        dataField: 'ping',
+        value: ONE_EGLD,
+        gasLimit: 20_000_000,
+      });
+      logMessage(`Called 'ping' function of the contract. Tx hash: ${pingTxHash}`);
+
+      const pongTxHash = await sendTransaction({
+        chainSimulatorUrl: CHAIN_SIMULATOR_URL,
+        sender: ALICE_ADDRESS,
+        receiver: scAddress,
+        dataField: 'pong',
+        value: '0',
+        gasLimit: 20_000_000,
+      });
+      logMessage(`Called 'pong' function of the contract. Tx hash: ${pongTxHash}`);
+    }
+
+    await new Promise(resolve => setTimeout(resolve, 2000));
+
+    let numOfEventsReceived = 0;
+    let lastProcessedTimestamp = 0;
+    let counter = 0;
+    while (true) {
+      await eventProcessor.start(new EventProcessorOptions({
+        emitterAddresses: [scAddress],
+        eventIdentifiers: ['ping', 'pong'],
+        getLastProcessedTimestamp: async () => lastProcessedTimestamp,
+        elasticUrl: ELASTIC_SEARCH_URL,
+        onEventsReceived: async (highestTimestamp, events) => {
+          logMessage(`event processor received ${events.length} events with the highest timestamp ${highestTimestamp}`);
+          numOfEventsReceived += events.length;
+        },
+        setLastProcessedTimestamp: async (timestamp) => {
+          lastProcessedTimestamp = timestamp;
+        },
+        pageSize: 7,
+        delayBetweenRequestsInMilliseconds: 100,
+      }));
+
+      await new Promise(resolve => setTimeout(resolve, 100));
+      logMessage(`Running event processor #${counter + 1}`);
+
+      counter++;
+      if (counter > 10 || numOfEventsReceived === 2 * numPingPongs) {
+        break;
+      }
+    }
+
+    expect(numOfEventsReceived).toBe(numPingPongs * 2);
+  }, 100000);
```

Comment:
* this timeout value is not to big ?

Reply:
it means 1min40s, so it's fine

Reply:
okay, just to be sure that the time is ok